File: EmployeeManager/EmployeeManager.Domain/Models/Employee.cs
using EmployeeManager.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace EmployeeManager.Domain.Models
{
	[Table("Employees")]
	public class Employee : BaseEntity
	{
		[Display(Name = "First Name"), StringLength(60, MinimumLength = 3, ErrorMessage = "{0} size should be between {2} and {1} characters.")]
		public required string FirstName { get; set; }

		[Display(Name = "Last Name"), StringLength(60, MinimumLength = 3, ErrorMessage = "{0} size should be between {2} and {1} characters.")]
		public required string LastName { get; set; }

		[DataType(DataType.EmailAddress), EmailAddress(ErrorMessage = "Enter a valid e-mail."), MaxLength(40)]
		public required string Email { get; set; }

		[Display(Name = "Document Number"), StringLength(20, MinimumLength = 6, ErrorMessage = "{0} size should be between {2} and {1} characters.")]
		public required string DocumentNumber { get; set; }

		public List<PhoneNumber>? PhoneNumbers { get; set; }

		[ForeignKey("ManagerId")]
		public virtual Employee? Manager { get; set; }
		public long? ManagerId { get; set; }

		[Display(Name ="Role")]
		public required ERoleType Role { get; set; }

		[Display(Name = "Birth Date"), DataType(DataType.Date), DisplayFormat(DataFormatString = "{0:dd/MM/yyyy}"), Required(AllowEmptyStrings = false, ErrorMessage = "Inform {0}.")]
		public required DateTime BirthDate { get; set; }

		public required bool Active { get; set; }

		public Employee() { }

		[SetsRequiredMembers]
		public Employee(string firstName, string lastName, string email, string documentNumber, List<PhoneNumber>? phoneNumbers, Employee? manager, ERoleType role, DateTime birthDate, bool active)
		{
			FirstName = firstName;
			LastName = lastName;
			Email = email;
			DocumentNumber = documentNumber;
			PhoneNumbers = phoneNumbers;
			Manager = manager;
			Role = role;
			BirthDate = birthDate;
			Active = active;
		}

		public bool IsMinor()
		{
			return BirthDate.AddYears(18).Date > DateTime.Now.Date;
		}

		public bool CanBeManager(ERoleType employeeRole)
		{
			return (int)employeeRole >= (int)Role;
		}
	}
}

File: EmployeeManager-Client/src/app/shared/models/employee.ts
import { ERoleType } from '../enums/e-role-type';
import { PhoneNumber } from './phone-number';

export class Employee {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  documentNumber: string;
  phoneNumbers: PhoneNumber[];
  managerId: number | null;
  role: ERoleType;
  birthDate: Date;
  active: boolean;

  //#region CUSTOM

  name: string | null;
  password: string;

  //#endregion

  constructor(
    id: number,
    firstName: string,
    lastName: string,
    email: string,
    documentNumber: string,
    phoneNumbers: PhoneNumber[],
    managerId: number,
    role: ERoleType,
    birthDate: Date,
    active: boolean,
    name: string,
    password: string
  ) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.documentNumber = documentNumber;
    this.phoneNumbers = phoneNumbers;
    this.managerId = managerId;
    this.role = role;
    this.birthDate = birthDate;
    this.active = active;
    this.name = name;
    this.password = password;
  }
}
